### PR TITLE
Incubo4u/ex-228/mapping Bug fix

### PR DESCRIPTION
**Pull request by incubo4u**

### CODE REVIEW

File: example/src/components/RoomParticipant.tsx [hunk 1]
```diff
@@ -56,12 +56,8 @@ export const RoomParticipant = ({
...
   const buttonOpacity = useSharedValue(0);
 
-  const participantHasVideo = () => {
-    if (videoTrack) {
-      return videoTrack.metadata.active;
-    }
-    return false;
-  };
+  const participantHasVideo = () =>
+    videoTrack ? videoTrack.metadata.active : false;
```
Comment (karkakol) [inline example/src/components/RoomParticipant.tsx:60]: videoTrack && videoTrack.metadata.active does the same but shorter :)

File: ios/MembraneWebRTCModule.swift [hunk 2]
```diff
@@ -104,7 +104,7 @@ public class MembraneWebRTCModule: Module {
       try membraneWebRTC.toggleMicrophone()
     }
     
-    Property("IsCameraOn") {
+    Property("isCameraOn") {
```
Comment (karkakol) [inline ios/MembraneWebRTCModule.swift:107]: Why here camelCase and in android PascalCase?

File: src/index.tsx [hunk 3]
```diff
@@ -398,12 +398,14 @@ export function useCamera() {
...
  */
 export function useMicrophone() {
-  const [isMicrophoneOn, setIsMicrophoneOn] = useState<boolean>(false);
+  const [isMicrophoneOn, setIsMicrophoneOn] = useState<boolean>(
+    MembraneWebRTCModule.isMicrophoneOn
+  );
 
   useEffect(() => {
     const eventListener = eventEmitter.addListener<IsMicrophoneOnEvent>(
       'IsMicrophoneOn',
```
Comment (karkakol) [inline src/index.tsx:407]: I was wondering is it a good idea to create a script which generates swift, kotlin and react  files based on json config, that will get rid off discrepancy between events naming

config could look like:
{
  "MAMA" : "MAMA",
  "TATA" : "TATA",
  "BRAT" : "BRAT"
}
 and script ( in python for example):
 
 import json

with open('variables.json', 'r') as f:
    variables = json.load(f)

with open('variables.ts', 'w') as f:
    for key, value in variables.items():
        f.write(f'const {key} = "{value}";\n')

with open('variables.swift', 'w') as f:
    for key, value in variables.items():
        f.write(f'let {key} = "{value}"\n')

with open('variables.kt', 'w') as f:
    f.write('object Constants {\n')
    for key, value in variables.items():
        f.write(f'    val {key} = "{value}"\n')
    f.write('}\n')

File: android/src/main/java/com/reactnativemembrane/MembraneWebRTCModule.kt [hunk 4]
```diff
@@ -193,19 +193,19 @@ class MembraneWebRTCModule : Module() {
...
         }
 
         AsyncFunction("updateAudioTrackMetadata") Coroutine { metadata: Map<String, Any> ->
             withContext(Dispatchers.Main) {
-                membraneWebRTC.updateAudioTrackMetadata(metadata)
+                membraneWebRTC.updateLocalAudioTrackMetadata(metadata)
             }
         }
 
         AsyncFunction("updateScreencastTrackMetadata") Coroutine { metadata: Map<String, Any> ->
```
Comment (karkakol) [inline android/src/main/java/com/reactnativemembrane/MembraneWebRTCModule.kt:206]: Are you sure that discrepancy between AsyncFunction name and mambraneWebRTC.method is valid?

**Review by karkakol — CHANGES_REQUESTED** [review]

File: ios/MembraneWebRTCModule.swift (hunk 2 above)
Comment (incubo4u) [reply ios/MembraneWebRTCModule.swift:107]: https://github.com/jellyfish-dev/react-native-membrane-webrtc/blob/2184aa99fd8bf36207b9e4d31e66a4b3fc56badb/android/src/main/java/com/reactnativemembrane/MembraneWebRTCModule.kt#L144

i belive that on android camelCase is used

File: android/src/main/java/com/reactnativemembrane/MembraneWebRTCModule.kt (hunk 4 above)
Comment (incubo4u) [reply android/src/main/java/com/reactnativemembrane/MembraneWebRTCModule.kt:206]: Are you asking if the functions can have different names or if they should have different names? They can indeed have different names, and the function will still be correctly called. However, I agree that they should ideally have the same name. I opted to change the name for clarity in the native code and didn't want to introduce a breaking change to React Native

Comment (incubo4u) [reply android/src/main/java/com/reactnativemembrane/MembraneWebRTCModule.kt:206]: If you believe this is a bad idea, please let me know 😊

File: ios/MembraneWebRTC.swift [hunk 5]
```diff
@@ -8,840 +8,883 @@ import ExpoModulesCore
...
+                    return
+                }
+                do {
+                    guard let screencastTrack = localScreencastTrack else {return }
+                    try removeTrackFromLocalEndpoint(screencastTrack)
+                    localScreencastTrack = nil
+                    try setScreencastTrackState(isEnabled: false)
+                }
+                catch{
+                    print("Error stoping screencast: \(error)")
```
Comment (karkakol) [inline ios/MembraneWebRTC.swift:442]: why print and not logger?

File: ios/MembraneWebRTC.swift [hunk 6]
```diff
@@ -8,840 +8,883 @@ import ExpoModulesCore
...
+        try ensureConnected()
+        guard let screencastExtensionBundleId = Bundle.main.infoDictionary?["ScreencastExtensionBundleId"] as? String else {
+            throw Exception(name: "E_NO_BUNDLE_ID_SET", description: "No screencast extension bundle id set. Please set ScreencastExtensionBundleId in Info.plist")
+        }
+        guard let appGroupName = Bundle.main.infoDictionary?["AppGroupName"] as? String else {
+            throw Exception(name: "E_NO_APP_GROUP_SET", description: "No app group name set. Please set AppGroupName in Info.plist")
+        }
+        // if screensharing is enabled it must be closed by the Broadcast Extension, not by our application
+        // the only thing we can do is to display stop recording button, which we already do
+        guard isScreensharingEnabled == false else {
```
Comment (karkakol) [inline ios/MembraneWebRTC.swift:398]: why not !isScreensharingEnabled ?

File: ios/MembraneWebRTC.swift [hunk 7]
```diff
@@ -1,844 +1,883 @@
...
+    }
+    private func removeTrackFromLocalEndpoint(_ track: LocalVideoTrack) throws {
+        try ensureEndpoints()
+        if let localEndpointId = localEndpointId {
+            guard let localEndpoint = MembraneRoom.sharedInstance.endpoints[localEndpointId] else { return }
+            MembraneRoom.sharedInstance.endpoints[localEndpointId] = localEndpoint.removeTrack(trackId: track.trackId())
+            emitEndpoints()
+        }
+    }
+    private func removeTrackFromLocalEndpoint(_ track: LocalAudioTrack){}
```
Comment (kamil-stasiak) [inline ios/MembraneWebRTC.swift:331]: Why is this function empty?

File: android/src/main/java/com/reactnativemembrane/MembraneWebRTC.kt [hunk 8]
```diff
@@ -609,23 +655,23 @@ class MembraneWebRTC(val sendEvent: (name: String, data: Map<String, Any?>) -> U
...
-        audioDevices: List<AudioDevice>,
-        selectedDevice: AudioDevice?
+        audioDevices: List<AudioDevice>, selectedDevice: AudioDevice?
     ) {
-        val map = mapOf(
-            "selectedDevice" to (if (selectedDevice != null) audioDeviceAsRNMap(selectedDevice) else null),
-            "availableDevices" to audioDevices.map { audioDevice -> audioDeviceAsRNMap(audioDevice) }
-        )
+        val map =
+            mapOf("selectedDevice" to (if (selectedDevice != null) audioDeviceAsRNMap(selectedDevice) else null),
```
Comment (kamil-stasiak) [inline android/src/main/java/com/reactnativemembrane/MembraneWebRTC.kt:661]: The main key of this map is `selectedDevice` and the event id is `AudioDeviceUpdate`. Shouldn't they be the same?

**Review by kamil-stasiak — CHANGES_REQUESTED** [review]
This code is readeble

File: ios/MembraneWebRTC.swift [hunk 9]
```diff
@@ -8,837 +8,883 @@ import ExpoModulesCore
...
+    //  func connect(metadata: [String: Any], promise: Promise) -> Void {
+    //      connectPromise = promise
+    //      .localUserMetadata = metadata.toMetadata()
+    //
+    //      if let localEndpointId = localEndpointId {
+    //          .endpoints[localEndpointId]?.metadata = metadata.toMetadata()
+    //    }
+    //
+    //    membraneRTC?.connect(metadata: metadata.toMetadata())
+    //  }
```
[the remark this answers is not in the document]
Comment (incubo4u) [reply ios/MembraneWebRTC.swift]: nope, removing

File: android/src/main/java/com/reactnativemembrane/MembraneWebRTC.kt (hunk 8 above)
Comment (incubo4u) [reply android/src/main/java/com/reactnativemembrane/MembraneWebRTC.kt:661]: there are 2 main keys in this map

**Review by kamil-stasiak — APPROVED** [review]

File: example/src/components/RoomParticipant.tsx [hunk 10]
```diff
@@ -139,11 +135,11 @@ export const RoomParticipant = ({
     }
     return null;
   };
-
   return (
     <View style={styles.fill}>
       <Pressable onPress={triggerShowingPinButton} style={styles.fill}>
         {participantHasVideo() ? (
+          // @ts-ignore
```
Comment (karkakol) [inline example/src/components/RoomParticipant.tsx:142]: czemu ts-ignore?
barfdo nieładnie

File: example/src/screens/Preview.tsx [hunk 11]
```diff
@@ -103,6 +103,7 @@ export const Preview = ({ navigation, route }: Props) => {
 
             <View style={styles.cameraPreview}>
               {isCameraOn ? (
+                // @ts-ignore
```
Comment (karkakol) [inline example/src/screens/Preview.tsx:106]: dlaczemu ts-ignore?

**Pull request merged**